src/state.rs

```diff
@@ -164,6 +173,16 @@ impl State {
         Ok(Self { tx })
     }
 
+    fn overall_health(connected_rpcs: Vec<RpcInfo>) -> Health {
```

Comment:
If all RPCs are OK, ServerHealth is OK,
if not all OK but some, ServerHealth is Degraded
in none are OK, ServerHealth is Critical